### PR TITLE
[CLOUDTRUST-3966] Adapt bridge for vouchers

### PR DESCRIPTION
**Pull request by basbeu**

### CODE REVIEW

File: internal/keycloakb/accountingclient.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+type AccountingClient struct {
+	httpClient HttpClient
+}
+
+type HttpClient interface {
+	Get(data interface{}, plugins ...plugin.Plugin) error
+}
+
+type AccountingBalance struct {
+	Balance *float32 `json:"balance"`
```
Comment (fperot74) [inline internal/keycloakb/accountingclient.go:30]: float64 is used in accounting (but I don't think it can cause an issue with our values)

File: internal/keycloakb/accountingclient.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+func MakeAccountingClient(httpClient HttpClient) *AccountingClient {
+	return &AccountingClient{
+		httpClient: httpClient,
+	}
+}
+
+func (c *AccountingClient) GetBalance(ctx context.Context, realmName string, userID string, service string) (float32, error) {
+	var correlationID = ctx.Value(cs.CtContextCorrelationID).(string)
+	var res AccountingBalance
+	err := c.httpClient.Get(&res, url.Path(getBalancePath), url.Param(prmRealmName, realmName), url.Param(prmUserID, userID), url.Param(prmService, service), headers.Set(hdrCorrID, correlationID))
```
Comment (fperot74) [inline internal/keycloakb/accountingclient.go:42]: Could you test it?
Query parameter should be added with query.Add("service", service)... (and it is not part of the URL)

File: pkg/mobile/component.go [hunk 3]
```diff
@@ -146,6 +152,16 @@ func (c *component) GetUserInformation(ctx context.Context) (api.UserInformation
 			c.logger.Debug(ctx, "msg", "User is not allowed to access video identification", "id", ctx.Value(cs.CtContextUserID))
 			delete(availableChecks, actionIDNow)
 		}
+
+		// if vouchers are required to proceed to a video identification, check that the balance is bigger than 1
+		if realmAdminConfig.VideoIdentificationAccountingEnabled != nil && realmAdminConfig.VideoIdentificationPrepaymentRequired != nil &&
```
Comment (fperot74) [inline pkg/mobile/component.go:157]: No need to enter this `if` statement if `actionIDNow` is not in `availableChecks` (avoid 1 HTTP query and associated SQL query behind it)
You also can write a function to be able to write `realmAdminConfig.IsIDNowActivated()` and hide this long checks

**Review by fperot74 — APPROVED** [review]

File: internal/keycloakb/accountingclient.go (hunk 1 above)
Comment (basbeu) [reply internal/keycloakb/accountingclient.go:30]: I will change to have the same type here !

File: internal/keycloakb/accountingclient.go (hunk 2 above)
Comment (basbeu) [reply internal/keycloakb/accountingclient.go:42]: I tested it ! It works but well I think it is better to do it with  query.Add("service", service) to make it clean.

File: pkg/mobile/component.go (hunk 3 above)
Comment (basbeu) [reply pkg/mobile/component.go:157]: Good catch !

File: internal/keycloakb/accountingclient.go [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+	"context"
+
+	cs "github.com/cloudtrust/common-service/v2"
+	"gopkg.in/h2non/gentleman.v2/plugin"
+	"gopkg.in/h2non/gentleman.v2/plugins/headers"
+	"gopkg.in/h2non/gentleman.v2/plugins/url"
+)
+
+const (
+	getBalancePath = "/realms/:realm/users/:userID/balance?service=:service"
```
Comment (fperot74) [inline internal/keycloakb/accountingclient.go:13]: Missing /internal at the beggining of the path

File: internal/keycloakb/accountingclient_test.go [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+		assert.Nil(t, err)
+		assert.Equal(t, expectedBalance, balance)
+	})
+
+	t.Run("FAILURE", func(t *testing.T) {
+		mockHttpClient.EXPECT().Get(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(expectedError).Times(1)
+
+		balance, err := accountingClient.GetBalance(ctx, realmName, userID, service)
+		assert.NotNil(t, err)
+		assert.Equal(t, float32(0), balance)
```
Comment (fperot74) [inline internal/keycloakb/accountingclient_test.go:48]: You don't need to test returned value when err is not nil

File: api/account/api.go [hunk 6]
```diff
@@ -77,17 +77,20 @@ type CredentialRepresentation struct {
...
+	ShowPasswordTab                       *bool           `json:"show_password_tab"`
+	ShowProfileTab                        *bool           `json:"show_profile_tab"`
+	ShowAccountDeletionButton             *bool           `json:"show_account_deletion_button"`
+	RedirectSuccessfulRegistrationURL     *string         `json:"redirect_successful_registration_url"`
+	AvailableChecks                       map[string]bool `json:"available-checks"`
+	BarcodeType                           *string         `json:"barcode_type"`
+	Theme                                 *string         `json:"theme"`
+	SupportedLocales                      *[]string       `json:"supportedLocales,omitempty"`
+	ShowGlnEditing                        *bool           `json:"show_gln_editing,omitempty"`
+	VoucherEnabled                        *bool           `json:"voucher_enabled"`
```
Comment (harture) [inline api/account/api.go:91]: Do we have to expose in account API those 3 parameters ? voucher-enabled is not enough ?

File: pkg/mobile/component.go [hunk 7]
```diff
@@ -146,6 +152,16 @@ func (c *component) GetUserInformation(ctx context.Context) (api.UserInformation
 			c.logger.Debug(ctx, "msg", "User is not allowed to access video identification", "id", ctx.Value(cs.CtContextUserID))
 			delete(availableChecks, actionIDNow)
 		}
+
+		// if vouchers are required to proceed to a video identification, check that the balance is bigger than 1
+		if _, ok := availableChecks[actionIDNow]; ok && realmAdminConfig.VideoIdentificationAccountingEnabled != nil && realmAdminConfig.VideoIdentificationPrepaymentRequired != nil &&
+			*realmAdminConfig.VideoIdentificationAccountingEnabled && *realmAdminConfig.VideoIdentificationPrepaymentRequired {
+			balance, err := c.accountingClient.GetBalance(ctx, realm, userID, "VIDEO_IDENTIFICATION")
```
Comment (harture) [inline pkg/mobile/component.go:159]: log the error

**Review by harture — APPROVED** [review]
